mod/pages/views/default/resources/pages/edit.php

```diff
@@ -1,44 +1,53 @@
 <?php
+
 /**
  * Edit a page
  *
  * @package ElggPages
  */
-
 elgg_gatekeeper();
 
-$page_guid = (int) elgg_extract('guid', $vars);
+$page_guid = (int)elgg_extract('guid', $vars);
 $page = get_entity($page_guid);
-if (!pages_is_page($page)) {
+
+if (!pages_is_page($page) || !$page->canEdit()) {
 	register_error(elgg_echo('noaccess'));
-	forward('');
+	forward('', '403');
 }
 
 $container = $page->getContainerEntity();
 if (!$container) {
 	register_error(elgg_echo('noaccess'));
-	forward('');
+	forward('', '403');
+}
+
+elgg_set_page_owner_guid($container->guid);
```

Comment:
Why? As far as I am concerned nothing should be happening in the page handler - they should be gone altogether.